workers/dotnet/Temporalio.Omes/KitchenSinkWorkflow.cs

```diff
@@ -0,0 +1,387 @@
+
+    private readonly Queue<ActionSet> _actionSetQueue = new();
+
+    [WorkflowSignal("do_actions_signal")]
+    public async Task DoActionsSignal(DoSignal.Types.DoSignalActions doSignals)
+    {
+        if (doSignals.DoActionsInMain is { } inMain)
+        {
+            _actionSetQueue.Enqueue(inMain);
+        }
+        else
+        {
+            await HandleActionSet(doSignals.DoActions);
+        }
+    }
+
+    [WorkflowUpdate("do_actions_update")]
+    public async Task<object> DoActionsUpdate(DoActionsUpdate actionsUpdate)
+    {
+        var retval = await HandleActionSet(actionsUpdate.DoActions);
+        if (retval != null)
+        {
+            return retval;
+        }
+
+        return _workflowState;
+    }
+
+    [WorkflowUpdateValidator(nameof(DoActionsUpdate))]
+    public void DoActionsUpdateValidator(DoActionsUpdate actionsUpdate)
+    {
+        if (actionsUpdate.RejectMe != null)
+        {
+            throw new ApplicationFailureException("Rejected");
+        }
+    }
+
+    [WorkflowQuery("report_state")]
+    public WorkflowState ReportState(object _) => _workflowState;
+
+    [WorkflowRun]
+    public async Task<Payload?> RunAsync(WorkflowInput? workflowInput)
+    {
+        // Run all initial input actions
+        if (workflowInput?.InitialActions is { } actions)
+        {
+            foreach (var actionSet in actions)
+            {
+                var returnMe = await HandleActionSet(actionSet);
+                if (returnMe != null)
+                {
+                    return null;
+                }
+            }
+        }
+
+        // Run all actions from signals
+        while (true)
+        {
+            await Workflow.WaitConditionAsync(() => _actionSetQueue.Count > 0);
+            var actionSet = _actionSetQueue.Dequeue();
+            var returnMe = await HandleActionSet(actionSet);
+            if (returnMe != null)
+            {
+                return returnMe;
+            }
+        }
+    }
+
+    private async Task<Payload?> HandleActionSet(ActionSet actionSet)
+    {
+        Payload? returnMe = null;
+        // If actions are non-concurrent, just execute and return if requested
+        if (!actionSet.Concurrent)
+        {
+            foreach (var action in actionSet.Actions)
+            {
+                returnMe = await HandleAction(action);
+                if (returnMe != null)
+                {
+                    return returnMe;
+                }
+            }
+
+            return returnMe;
+        }
+
+        // If actions are concurrent, run them all, returning early if any action wishes to return
+        var tasks = new List<Task>();
+        foreach (var action in actionSet.Actions)
+        {
+            async void Action()
+            {
+                returnMe = await HandleAction(action);
+            }
+
+            var task = new Task(Action);
+            task.Start();
+            tasks.Add(task);
```

Comment:
I think it does work but the IDE wanted to complain about it. Not entirely sure why, since what's the point of a nice async closure if you're not supposed to use it heh.

Reply:
Actually it doesn't work to have it fully inlined to `Add`, you still have to make the new task. The explanation [here](https://www.jetbrains.com/help/rider/AsyncVoidLambda.html) is reasonable too so I'll probably just keep this as-is.